src/importexport/musicxml/internal/musicxml/importmxmlpass1.cpp

```diff
@@ -780,17 +780,17 @@ static void inferFromTitle(String& title, String& inferredSubtitle, String& infe
     StringList subtitleLines;
     StringList creditLines;
     StringList titleLines = title.split(std::regex("\\n"));
-    for (int i = titleLines.size() - 1; i > 0; --i) {
+    for (size_t i = titleLines.size() - 1; i > 0; --i) {
```

Comment:
We should make really sure that `titleLines.size()` is not `0`, because when it is, it's gonna be a very long loop :)(since `size_t(-1) > 0`).

Also, @miiizen, are you sure you didn't mean `i >= 0`? (If you did, then `size_t` would break this stop condition completely)

Reply:
Oops, check again, it should now fix the wanings **and** the logic ;-(

Reply:
Hmm, maybe not, see https://github.com/musescore/MuseScore/actions/runs/8432214967/job/23090896406?pr=22056#step:9:11312

Reply:
Cant we replace logic from lines 780 to 803 to 

`
StringList subtitleLines;
StringList creditLines;
StringList titleLines = title.split(std::regex("\\n"));

for (size_t i = titleLines.size(); i > 0; --i) {
    String line = titleLines[i - 1];

    if (isLikelyCreditText(line, true)) {
        creditLines.push_front(line);
        titleLines.erase(titleLines.begin() + i - 1);
    } else if (isLikelySubtitleText(line, true)) {
        subtitleLines.push_front(line);
        titleLines.erase(titleLines.begin() + i - 1);
    }
}

title = titleLines.join(u"\n");
inferredSubtitle = subtitleLines.join(u"\n");
inferredCredits = creditLines.join(u"\n");
`

this way we can use single loop without much iteration and without much memory waste
Maybe i am wrong but any advice would be really appreciated

Reply:
there's no `push_front()` method

Reply:
but `ìnsert(0, ...)`

Reply:
Seems to solve it nicely ;-)

Reply:
It seems to cause #24553 though

Reply:
Actually a simple off-by-one error, see #24555